scripts/generate-shaders.js

```diff
@@ -82,18 +81,21 @@ ${fragmentPrelude}
                     "{name} = unpack_mix_{a_type}(a_{name}, a_{name}_t);"
                 ]
             });
-    }
 
-    function fragmentSource() {
-        const source = fs.readFileSync(path.join(inputPath, shaderName + '.fragment.glsl'), 'utf8');
-        return applyPragmas(source, {
+    const fragmentSource = applyPragmas(fs.readFileSync(path.join(inputPath, shaderName + '.fragment.glsl'), 'utf8'), {
                 define: [
                     "varying {precision} {type} {name};"
                 ],
                 initialize: [
                 ]
             });
-    }
+
+    const shasum = crypto.createHash('sha1');
```

Comment:
I considered that as well, but it means that we'd have to expose some hashing function to C++ land via platform bindings, and incur the run-time cost of hashing. If you think it's safer, I'll go back and add it.